timedctl/timedctl.py

```diff
@@ -480,19 +479,45 @@ def generate_timesheet():
                 duration = attr["to-time"] - attr["from-time"]
                 # get task
                 task = activity_obj["relationships"]["task"]["data"]["id"]
-                # create report
-                timed.reports.post(
-                    {
-                        "duration": duration,
-                        "comment": attr["comment"],
-                    },
-                    {"task": task},
-                )
+                # check if there is a report with the same comment that already exists
+                report = [
+                    x
+                    for x in reports
+                    if x["attributes"]["comment"] == attr["comment"]
```

Comment:
the upstream implementation not only checks the comment but also other attr like the project.

this helps ensure that you can use the same comment on the same day for different customers

Reply:
look one line below, I also look at the task id; every task belongs to one unique project

Reply:
that does seem to make sense, make me wonder about this:
https://github.com/adfinis/timed-frontend/blob/4096c58c20f74606bad0d15d70c30420e7d4e51b/app/index/activities/controller.js#L198-L206

maybe the frontend checks more stuff because it might not be stored on the server yet? i'm not sure tbh

Reply:
```py
               r.get("review") === data.review && 
               r.get("notBillable") === data.notBillable && 
               !r.get("verfiedBy") && 
               !r.get("isDeleted")
```

ig we can add the review and non billable checks but I don't think such an edge-case would ever occur.

same goes for already verified reports; if one is verified on one day, why would anyone create a second activity.

After all this seems to me like very bad client-side validation

Reply:
@hairmare can we just merge this and move the TBD stuff into an issue so it's usable?

Reply:
up to you i guess ;)

Reply:
approve plz